src/pages/index.js

```diff
@@ -0,0 +1,257 @@
+/*-----------------------------------------------------------------*/
+
+const validationSettings = {
+  inputSelector: ".modal__input",
+  submitButtonSelector: ".modal__button",
+  inactiveButtonClass: "modal__button_disabled",
+  inputErrorClass: "modal__input_type_error",
+  errorClass: "modal__error_visible",
+};
+
+const editProfileFormValidator = new FormValidator(
+  validationSettings,
+  editProfileForm
+);
+editProfileFormValidator.enableValidation();
+
+const addCardFormValidator = new FormValidator(
+  validationSettings,
+  addCardFormElement
+);
+addCardFormValidator.enableValidation();
+/*-----------------------------------------------------------------*/
+/*                             New Instances                       */
+/*-----------------------------------------------------------------*/
+
+//New Image Popup
+const newImagePopup = new PopupWithImage("#view-card-modal");
+newImagePopup.setEventListeners();
+function renderCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  return card.getView();
+}
+
+function handleLikeClick(card) {
+  const action = card._isLiked
+    ? api.removeLike.bind(api)
+    : api.likeCard.bind(api);
+  action(card._id)
+    .then((res) => {
+      card.updateLikes(res.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+// need an if/else statement that checks whether the card is currently liked or not.
+//hint: isLiked is true then we want to tell the server to remove a like
+// else we tell the server to remove the like
+// when finished, (inside .then) update view of heart button
+
+//New Card Popup
+const newCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+newCardPopup.setEventListeners();
+addNewCardButton.addEventListener("click", () => {
+  addCardFormValidator.resetValidation();
+  newCardPopup.openModal();
+});
+
+//Profile Edit Popup
+const newProfileEdit = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+newProfileEdit.setEventListeners();
+
+//Avatar Edit Popup
+const newAvatarEdit = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+avatarEditButton.addEventListener("click", () => {
+  newAvatarEdit.openModal();
+});
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  editProfileForm
+);
+
+/*-----------------------------------------------------------------*/
+/*                   "Handle" Functions                            */
+/*-----------------------------------------------------------------*/
+
+function handleProfileEditSubmit(inputValues) {
+  newProfileEdit.setSaving(true);
+  console.log(inputValues);
+  api.editUserInfo(inputValues).then(() => {
+    userInfo.setUserInfo({
+      name: inputValues.title,
+      about: inputValues.description,
+    });
+    newProfileEdit.setSaving(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.

**Please, correct it everywhere**